### ISSUE TITLE
getCommitDetails problem on Azure

### ISSUE THREAD

**Issue by sylvain-combe-sonarsource** [SonarSource-Demos/sonar-golc#6]
For some projects the handleNonDefaultBranch method of the getazure package will fail.
This part of the code is responsible:

```
	branches, err := gitClient.GetBranches(ctx, git.GetBranchesArgs{
		RepositoryId: &repoID,
		Project:      &projectID,
	})
	if err != nil {
		return "", 0, 0, err
	}

```
An error there breaks the entire script, it should be handled gracefully.

**Comment by pstember** [comment]
Can you try with the version on this [branch](https://github.com/SonarSource-Demos/sonar-golc/tree/bugfix/azure-issue-6)?
I have added an error log to get more insights.

```go
	branches, err := gitClient.GetBranches(ctx, git.GetBranchesArgs{
		RepositoryId: &repoID,
		Project:      &projectID,
	})
	if err != nil {
		loggers.Errorf("Error get GetBranches for repository and branch %s - %s: %v", repoID, defaultBranch, err)
		return "", 0, 0, err
	}
``` 

Once we understand the problem and where to gracefully handle it, we will still need a fix.

We could also add a trace log before each function is called, to have the full path followed by the code.
This would take more time but can be done if needed, from there, it should be easy to handle the error

**Comment by sylvain-combe-sonarsource** [comment]
![image](https://github.com/user-attachments/assets/1eb85377-3734-461c-a23b-2f096bf9382c)

I believe there is no point logging the dafultBranch parameter in the handleNonDefaultBranch method.

Your comment's code (I don't see your commit) does not handle errors more gracefully than the existing code.

**Comment by pstember** [comment]
Indeed, the goal is to gather more insights to understand the problem, so the number of logs in the error should increase.
Handling the error gracefully is the next step once we understand what the error is.

**Comment by fabio-gos-sonarsource** [comment]
This is an old issue. I'm closing it as it seems to not be applicable anymore.
